visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -338,8 +327,7 @@ export class CodeMapMouseEventService implements ViewCubeEventPropagationSubscri
 			CodeMapMouseEventService.changeCursorIndicator(CursorType.Grabbing)
 		}
 		this.mouseOnLastClick = { x: event.clientX, y: event.clientY }
-		this.unhoverBuilding()
-		$(document.activeElement).blur()
+		;(document.activeElement as HTMLElement).blur()
```

Comment:
Why is a semicolon at the beginning of this line?

Reply:
This is a JS specific thing: JS does not require semicolons at the end of statements and implicitly tries to intelligently place them between statements.
Few statements require a semicolon, otherwise the output is ambiguous and that's also the case here. This would be interpreted as function call otherwise. Semicolons may be inserted anywhere between the statements. No matter if it's at the end of a statement, in-between or before the next statement (as done here).

To give you an example:

```js
const abc = [1,2,3]
(() => console.log(abc))()
// TypeError: [1,2,3] is not a function
```

```js
const abc = [1,2,3]
;(() => console.log(abc))()
// [ 1, 2, 3 ]
```

Reply:
Thanks for your nice explanation!

Reply:
I checked again and you were right, that it's not necessary in this specific case. I removed it due to that.

Reply:
It is in fact added by our linter, so let's keep it as is.